pkg/runtime/grpc/info.go

```diff
@@ -18,41 +18,25 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"net"
 	"time"
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials/insecure"
-
 	"github.com/inspektor-gadget/inspektor-gadget/internal/deployinfo"
 	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
 )
 
-func loadRemoteDeployInfo() (*deployinfo.DeployInfo, error) {
+func (r *Runtime) loadRemoteDeployInfo(params *params.Params) (*deployinfo.DeployInfo, error) {
 	ctx, cancelDial := context.WithTimeout(context.Background(), time.Second*ConnectTimeout)
 	defer cancelDial()
-
-	// Get a random gadget pod and get the info from there
-	pods, err := getGadgetPods(ctx, []string{})
-	if err != nil {
-		return nil, fmt.Errorf("get gadget pods: %w", err)
+	if params == nil {
+		// use defaults, if params are unset (e.g. automatic fetching
+		// of catalog after deployment)
+		params = r.ParamDescs().ToParams()
 	}
```

Comment:
I can't understand it. nil is passed in all places where it's called, when a value other than nil will make sense?

Reply:
Hmm, I think for now we can use the default value for it, so I'll remove the parameter - later on it might make sense to fetch the deploy information from a specific host/node (e.g. if using a tcp socket), but we don't have a UI for that right now.